cmd/registry/cmd/upload/bulk/openapi.go

```diff
@@ -166,112 +165,82 @@ func (task *uploadOpenAPITask) populateFields() error {
 }
 
 func (task *uploadOpenAPITask) createAPI(ctx context.Context) error {
-	if _, err := task.client.GetApi(ctx, &rpc.GetApiRequest{
-		Name: task.apiName(),
-	}); !core.NotFound(err) {
-		return err // Returns nil when API is found without error.
-	}
-
-	response, err := task.client.CreateApi(ctx, &rpc.CreateApiRequest{
-		Parent: task.projectName() + "/locations/global",
-		ApiId:  task.apiID,
+	// Create an API if needed (or update an existing one)
+	response, err := task.client.UpdateApi(ctx, &rpc.UpdateApiRequest{
 		Api: &rpc.Api{
+			Name:        task.apiName(),
 			DisplayName: task.apiID,
 		},
+		AllowMissing: true,
 	})
 	if err == nil {
-		log.Debugf("Created %s", response.Name)
-	} else if core.AlreadyExists(err) {
-		log.Debugf("Found %s", task.apiName())
+		log.Debugf("Updated %s", response.Name)
 	} else {
 		log.WithError(err).Debugf("Failed to create API %s", task.apiName())
+		// Returning this error ends all tasks, which seems appropriate to
+		// handle situations where all might fail due to a common problem
+		// (a missing project or incorrect project-id).
```

Comment:
Does this comment belong on every error returned by tasks? Does this type of task especially depend on this behavior? From my perspective it seems like it's documentation for the task pool.

Reply:
That's a good point - looking back at the code, I recall putting this here in an attempt to explain why createAPI returns errors but not createVersion or createSpec (at some point, we might decide that they should too). I'll leave these in place but am open to alternatives and future changes.